seed/api/v2_1/views.py

```diff
@@ -122,3 +125,89 @@ def building_sync(self, request, pk):
         """
         return JsonResponse(
             {"status": "error", "message": "Not yet implemented. PK was {}".format(pk)})
+
+    @detail_route(methods=['PUT'])
+    def update_with_building_sync(self, request, pk):
+        """
+        Does not work in Swagger!
+
+        Update an existing PropertyView with a building file. Currently only supports BuildingSync.
+        ---
+        consumes:
+            - multipart/form-data
+        parameters:
+            - name: pk
+              description: The PropertyView to update with this buildingsync file
+              type: path
+              required: true
+            - name: organization_id
+              type: integer
+              required: true
+            - name: cycle_id
+              type: integer
+              required: true
+            - name: file_type
+              type: string
+              enum: ["Unknown", "BuildingSync", "GeoJSON"]
+              required: true
+            - name: file
+              description: In-memory file object
+              required: true
+              type: file
+        """
+        if len(request.FILES) == 0:
+            return JsonResponse({
+                'success': False,
+                'message': "Must pass file in as a Multipart/Form post"
+            })
+
+        the_file = request.data['file']
+        file_type = BuildingFile.str_to_file_type(request.data.get('file_type', 'Unknown'))
+        organization_id = request.data['organization_id']
+        cycle = request.data.get('cycle_id', None)
+
+        if not cycle:
+            return JsonResponse({
+                'success': False,
+                'message': "Cycle ID is not defined"
+            })
+        else:
+            cycle = Cycle.objects.get(pk=cycle)
+
+        building_file = BuildingFile.objects.create(
+            file=the_file,
+            filename=the_file.name,
+            file_type=file_type,
+        )
+
+        try:
+            # do I need to pass cycle ID to get a specific cycle time?
+            # and do I need to pass org ID to ensure orgs match?
+            property_view = PropertyView.objects.get(pk=pk)
+        except PropertyView.DoesNotExist:
+            return JsonResponse({
+                'success': False,
+                'message': 'Cannot match a PropertyView with pk=%s; cycle_id=%s' % (pk, cycle)
+            })
+
+        # here, instead of relying on BuildingFile.process to create a PropertyView,
+        # I'd like to either:
+        #   pass in the existing property_view and have it to the assignment inside process, or
+        #   just have the process() function only do the processing, and not also the PV creation, then I can do
+        #   the assignment here
```

Comment:
Most of this function was pulled from the `BuildingFileViewSet.create()` method.  We could certainly pull out some of the functionality into a single worker to maximize DRYness.  But for the sake of time, I dropped it over here and modified it.

Here is the core change from the create method.  The process() function used here does (at least) two things: 
 - processes the building sync XML file into a structure
 - creates the propertyview to accompany this buildingsync file

For the call in this view I'm adding, I only want the first step.  Would you prefer that I:
 - Modify that function so that the processing and creating are split into two functions, so that I can call the process method from here easier?
 - Modify that function so that I pass in an optional property view ID, and it will skip creating a new one?
 - Something else?

Reply:
look here for processing BuildingSync. https://github.com/SEED-platform/seed/blob/bricr/seed/models/building_file.py#L89

I think we would want to break up the processing/creation into two functions. Processing should be done'ish with the link above. Creating and updating the propertyview is the challenge as we need to keep all the relationships on the propertystate intact (measures, scenarios, etc).